dev-docs/source/Architecture.rst

```diff
@@ -0,0 +1,180 @@
+core functionality e.g.
+workspace objects, instrument objects, algorithm interfaces,
+along with a set of plugin libraries that are loaded at runtime to register
+new capabilities such as algorithms for data treatment.
+
+A Python package to the C++ libraries, called ``mantid``, is defined in
+``Framework/PythonInterface`` and
+gives access to the C++-framework code from Python.
+The Python package is the primary mode of access to the C++ libraries
+for external entities.
+While the C++ API could be used directly by external parties this is
+not a supported method of interaction and API stability of the C++
+layer is not guaranteed.
+
+mantidqt
+########
+
+.. figure:: /images/architecture-mantidqt.png
+   :scale: 70%
+   :align: center
+   :alt: Component diagram of mantidqt container
+
+   Container diagram for mantidqt Python package
+
+``mantidqt`` is a Python package defined house a collcetion of Qt widgets
+that sit on top of ``mantid`` functionality.
+They provide a reusable set of widgets that can be used across multiple
+interfaces to give a consistent user experience when presenting given
+``mantid`` functionality to users. One example of such a widget is the
+``mantidqt.widgets.filefinder``, which sits on top of the framework
+``FileFinder``, and presents GUI elements to pass through and run the
+the framework logic.
+Providing a reusable widget gives a consistent user experience across
+multiple interfaces/applications when displaying ``mantid`` features to users.
+
+The package contains a mixture of C++ and Python code.
+It is defined in the ``qt/python/mantidqt`` directory of the source tree.
+Widgets written in C++ use the `sip <sip-ref_>`_ system to export the
+C++ classes to a Python module.
+The raw exported modules are all prefixed with an underscore,
+e.g. ``_common``,
+and classes are pulled into a pure-python module to decouple
+the C++ implementations from users.
+This allows for flexibility with the layout of the C++ classes/libraries
+without impacting users in the future.
+Clients of ``mantidqt`` should never import an underscored module directly.
+
+workbench
+#########
+
+.. figure:: /images/architecture-workbench.png
+   :scale: 70%
+   :align: center
+   :alt: Component diagram of workbench container
+
+   Container diagram for workbench application
+
+``workbench`` is an application written in pure Python in the ``qt/applications/workbench``
+directory of the source.
+It is not intended to be a library imported by others and as such offers
+so guarantees of API stability.
+It makes heavy use of the ``mantidqt`` widgets collection and puts them
+together to build the application shipped to users as ``MantidWorkbench``.
+
+The application provides some widgets that are only applicable to itself,
+such as the about screen, settings screen and project recovery mechanism that
+is specific to workbench.
+
+Plotting within ``workbench`` uses matplotlib_.
```

Comment:
Shall we mention somewhere the OpenGL used in the instrument viewer? Or perhaps a small paragraph can mention that apart from plotting with matplotlib there are 2 advanced widgets: sliceviewer and instrument viewer.

Reply:
I might do this in the `mantidqt` and emphasize that it's not just basic thinks like text fields and combo boxes.